src/pages/user/register/index.tsx

```diff
@@ -0,0 +1,383 @@
+import { Form } from '@ant-design/compatible';
+import React, { Component } from 'react';
+import { Dispatch } from 'redux';
+import { FormComponentProps } from 'antd/es/form';
+import Link from 'umi/link';
+import { connect } from 'dva';
+import router from 'umi/router';
+
+import { StateType } from './model';
+import styles from './style.less';
+
+const FormItem = Form.Item;
+const { Option } = Select;
+const InputGroup = Input.Group;
+
+const passwordStatusMap = {
+  ok: (
+    <div className={styles.success}>
+      <FormattedMessage id="userandregister.strength.strong" />
+    </div>
+  ),
+  pass: (
+    <div className={styles.warning}>
+      <FormattedMessage id="userandregister.strength.medium" />
+    </div>
+  ),
+  poor: (
+    <div className={styles.error}>
+      <FormattedMessage id="userandregister.strength.short" />
+    </div>
+  ),
+};
+
+const passwordProgressMap: {
+  ok: 'success';
+  pass: 'normal';
+  poor: 'exception';
+} = {
+  ok: 'success',
+  pass: 'normal',
+  poor: 'exception',
+};
+
+interface RegisterProps extends FormComponentProps {
+  dispatch: Dispatch<any>;
+  userAndregister: StateType;
+  submitting: boolean;
+}
+interface RegisterState {
+  count: number;
+  confirmDirty: boolean;
+  visible: boolean;
+  help: string;
+  prefix: string;
+}
+
+export interface UserRegisterParams {
+  mail: string;
+  password: string;
+  confirm: string;
+  mobile: string;
+  captcha: string;
+  prefix: string;
+}
+
+@connect(
+  ({
+    userAndregister,
+    loading,
+  }: {
+    userAndregister: StateType;
+    loading: {
+      effects: {
+        [key: string]: string;
+      };
+    };
+  }) => ({
+    userAndregister,
+    submitting: loading.effects['userAndregister/submit'],
+  }),
+)
+class Register extends Component<RegisterProps, RegisterState> {
+  state: RegisterState = {
+    count: 0,
+    confirmDirty: false,
+    visible: false,
+    help: '',
+    prefix: '86',
+  };
+
+  interval: number | undefined = undefined;
+
+  componentDidUpdate() {
+    const { userAndregister, form } = this.props;
+    const account = form.getFieldValue('mail');
+    if (userAndregister.status === 'ok') {
+      message.success('注册成功！');
+      router.push({
+        pathname: '/user/register-result',
+        state: {
+          account,
+        },
+      });
+    }
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.interval);
+  }
+
+  onGetCaptcha = () => {
+    let count = 59;
+    this.setState({ count });
+    this.interval = window.setInterval(() => {
+      count -= 1;
+      this.setState({ count });
+      if (count === 0) {
+        clearInterval(this.interval);
+      }
+    }, 1000);
+  };
+
+  getPasswordStatus = () => {
+    const { form } = this.props;
+    const value = form.getFieldValue('password');
+    if (value && value.length > 9) {
+      return 'ok';
+    }
+    if (value && value.length > 5) {
+      return 'pass';
+    }
+    return 'poor';
+  };
+
+  handleSubmit = (e: React.FormEvent) => {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The class property 'handleSubmit' must be marked either 'private', 'public', or 'protected'](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4583897)